taskana-adapter-camunda-tasklistener/src/main/java/pro/taskana/adapter/camunda/tasklistener/TaskanaTaskListener.java

```diff
@@ -34,153 +33,223 @@ public static TaskanaTaskListener getInstance() {
         return instance;
     }
 
-    @Override
     public void notify(DelegateTask delegateTask) {
 
         try (Connection connection = Context.getProcessEngineConfiguration().getDataSource().getConnection()) {
 
+
             switch (delegateTask.getEventName()) {
 
                 case "create":
-                    insertCreateEvent(delegateTask, connection);
+                    insertCreateEventIntoOutbox(delegateTask, connection);
                     break;
                 case "complete":
-                    insertCompleteEvent(delegateTask, connection);
+                    insertCompleteOrDeleteEventIntoOutbox(delegateTask, connection);
                     break;
                 case "delete":
-                    insertDeleteEvent(delegateTask, connection);
+                    insertCompleteOrDeleteEventIntoOutbox(delegateTask, connection);
                     break;
             }
 
+        } catch (SQLException e) {
+            LOGGER.warn("Caught {} while trying to process a delegate task", e);
         } catch (Exception e) {
-
             LOGGER.warn("Caught {} while trying to process a delegate task", e);
         }
 
     }
 
-    private void insertCreateEvent(DelegateTask delegateTask, Connection connection) {
+    private void insertCreateEventIntoOutbox(DelegateTask delegateTask, Connection connection) throws SQLException {
+
+        String camundaSchema = null;
 
         try {
-            LOGGER.debug("TaskanaTaskListener uses Database with url {} ", connection.getMetaData().getURL());
 
-            String camundaSchema = connection.getSchema();
+            camundaSchema = connection.getSchema();
             LOGGER.debug("camundaSchema in taskListener is {}", camundaSchema);
-            String dbProductName = connection.getMetaData().getDatabaseProductName();
-            if ("PostgreSQL".equals(dbProductName)) {
-                connection.setSchema("taskana_tables");
-            } else {
-                connection.setSchema("TASKANA_TABLES");
+
+            setOutboxSchema (connection);
+
+            String referencedTaskJson = getReferencedTaskJson(delegateTask);
+
+            prepareAndExecuteStatement(connection, delegateTask, referencedTaskJson);
+            
+        } catch (SQLException e) {
+            LOGGER.warn("Caught {} while trying to insert a \"create\" event into the outbox table", e);
+
+        } finally {
+            if (camundaSchema!=null){
+                connection.setSchema(camundaSchema);
+            }
+        }
+    }
+
```

Comment:
Methods insertCreateEventIntoOutbox and insertCompleteOrDeleteEventIntoOutbox are almost identical -> there is an opportunity to refactor the methods to avoid duplicate code.
Not necessarily with this pull request. A ticket to do this with a future request would be sufficient.